File: core/src/main/java/io/tcc/core/model/enums/SubjectEnum.java
package io.tcc.core.model.enums;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.Arrays;

@RequiredArgsConstructor
@Getter
public enum SubjectEnum {

    NONE(0, "Sem Matéria"),
    MATEMATICA(1, "Matemática"),
    PORTUGUES(2, "Português"),
    BIOLOGIA(3, "Biologia"),
    QUIMICA(4, "Química"),
    FISICA(5, "Física"),
    GEOGRAFIA(6, "Geografia");

    private final Integer id;
    private final String name;

    public static SubjectEnum getByName(String name) {
        return Arrays.stream(SubjectEnum.values())
                .filter(subject -> subject.getName().equals(name)).findFirst().orElse(NONE);
    }
}

File: front/src/enums/SubjectEnum.ts
export class SubjectEnum {

  static NONE = new SubjectEnum(0, "Sem Matéria");
  static MATEMATICA = new SubjectEnum(1, "Matemática");
  static PORTUGUES = new SubjectEnum(2, "Português");
  static BIOLOGIA = new SubjectEnum(3, "Biologia");
  static QUIMICA = new SubjectEnum(4, "Química");
  static FISICA = new SubjectEnum(5, "Física");
  static GEOGRAFIA = new SubjectEnum(6, "Geografia");

  constructor(
    public value: number,
    public label: string
  ) {
    this.value = value;
    this.label = label
  }

  public static values(): SubjectEnum[] {
    return [
      this.NONE,
      this.MATEMATICA,
      this.PORTUGUES,
      this.BIOLOGIA,
      this.QUIMICA,
      this.FISICA,
      this.GEOGRAFIA
    ];
  }

}
